Cython/Compiler/Nodes.py

```diff
@@ -1147,7 +1147,17 @@ class MemoryViewSliceTypeNode(CBaseTypeNode):
 
     def analyse(self, env, could_be_name=False):
 
-        base_type = self.base_type_node.analyse(env)
+        if isinstance(self.base_type_node, CConstOrVolatileTypeNode):
+            # In this tree the 'const' qualifier of this memoryview is attached
+            # to the node of the base type, but its meaning is
+            # 'const memoryview'.  For non-PyObject types this doesn't make a
```

Comment:
I don't think the way `const` is handled for memory views is wrong. It means that the item data type is not writeable, which is correct. Think of structs, for example, where it's not just the memory view items but also the struct fields that must not be modified. Really, you could argue that there is no such thing as a `const object[:]`.

Reply:
Yes I agree that one can take this view.  For the use case I have in mind (see https://github.com/cython/cython/issues/2485#issuecomment-1079660938) though, the ultimate goal is to appease the numpy.nditer interpretation of the 'readonly' operand flag:  The operand ndarray is not written to, viz., no references in it are overwritten.  It doesn't make any assumptions on whether or not the referred to object changes.

Put differently:  Currently it is possible to appease the nditer/object ndarray/readonly combination by declaring the loop buffer through the legacy interface `cdef ndarray[object]`, but there is no way to achieve this with an `cdef object[:]` memoryview.  The proposed change is simply the most canonical solution from my perspective, and I would be happy with any other solution that would allow me to avoid the legacy API. (Without changing the 'readonly' flag to 'readwrite'.)